### PR TITLE
First reprocessing of health scanner data #81 [part 2]

### LINKED ISSUE #81: First reprocessing of health scanner data
update: running first dev pipeline tests with following changes (will appear formally in a changelog):
• ElasticsearchPlus developed to chain together row-wise transformations such as:
• “company” (for crunchbase) or “paper” (for nih) added as new field `type_of_entity`
• applying `null` values to fields as described in https://github.com/nestauk/nesta/blob/health2/nesta/production/schemas/tier_1/field_null_mappings/health_scanner.json
• Changed “crunchbase” to “companies” as `_index` name, in accordance with Ts&Cs of new crunchbase contract
• Aliases applied as described in https://github.com/nestauk/nesta/blob/health2/nesta/production/schemas/tier_1/aliases/health_scanner.json
• empty string filtered to `null`
• countries automatically detected in large text
• `terms_` fields are automatically listified, with the most likely delimiter automatically identified. In practice, this means the `;` seperated values will now be a proper json array from the ES response.
• Geographic lat lon values are now always floats

other changes:
• State names in CB data

Things not done yet (relatively short tasks) which will go into the same release:
⇒ Mixed analyzers for country and city lists
⇒ Add funders to CB and NiH
⇒ Add continent names

I’ll get in touch once the pipeline is flowing properly

### PR DESCRIPTION
**Pull request by jaklinger**
See the following issue for background:

Closes #81

### CODE REVIEW

File: nesta/production/orms/crunchbase_es_config.json [hunk 1]
```diff
@@ -1,129 +1,170 @@
...
+	    "number_of_shards": "5",
+	    "number_of_replicas": "1"
+	},
+	"analysis": {
+	    "analyzer": {
+		"mesh_terms_analyzer": {
+		    "type": "custom",
+		    "tokenizer": "standard",
+		    "filter": [
+			"standard",
```
Comment (mindrones) [inline nesta/production/orms/crunchbase_es_config.json]: to be indented

File: nesta/production/orms/crunchbase_es_config.json [hunk 2]
```diff
@@ -1,129 +1,170 @@
...
-                "terms_roles_organisation": {
-                    "type": "keyword"
-                },
-                "url_of_organisation": {
-                    "type": "keyword"
-                },
-                "placeName_country_organisation": {
-                    "type": "keyword"
-                },
+	"_doc": {
```
Comment (mindrones) [inline nesta/production/orms/crunchbase_es_config.json]: to be indented

File: nesta/production/orms/crunchbase_es_config.json [hunk 3]
```diff
@@ -1,129 +1,170 @@
...
+	    "dynamic": "strict",
+	    "properties": {
+		"name_of_organisation": {
```
Comment (mindrones) [inline nesta/production/orms/crunchbase_es_config.json]: - to be indented
- keys to be sorted alphabetically ignoring `_`

File: nesta/production/orms/crunchbase_es_config.json [hunk 4]
```diff
@@ -1,129 +1,170 @@
...
-            "number_of_shards": "5",
-            "number_of_replicas": "1"
-        }
+	"index": {
+	    "number_of_shards": "5",
+	    "number_of_replicas": "1"
+	},
+	"analysis": {
+	    "analyzer": {
+		"mesh_terms_analyzer": {
```
Comment (mindrones) [inline nesta/production/orms/crunchbase_es_config.json]: to be indented

File: nesta/production/orms/nih_es_config.json [hunk 5]
```diff
@@ -28,23 +28,46 @@
                     "type": "integer"
                 },
                 "placeName_city_organisation": {
```
Comment (mindrones) [inline nesta/production/orms/nih_es_config.json]: keys to be sorted alphabetically ignoring `_`

File: nesta/production/orms/nih_es_config.json [hunk 6]
```diff
@@ -28,23 +28,46 @@
...
                 "placeName_country_organisation": {
-                    "type": "keyword"
+                    "type": "text",
+                    "fields": {
+                        "keyword": {
+                            "type": "keyword"
+                        }
+                    }
                 },
 		"placeName_continent_organisation": {
```
Comment (mindrones) [inline nesta/production/orms/nih_es_config.json]: to be indented

File: nesta/production/orms/nih_es_config.json [hunk 7]
```diff
@@ -28,23 +28,46 @@
...
+                        }
+                    }
                 },
                 "placeName_zipcode_organisation": {
                     "type": "keyword"
                 },
                 "title_of_organisation": {
                     "type": "text"
                 },
+		"title_of_funder": {
```
Comment (mindrones) [inline nesta/production/orms/nih_es_config.json]: to be indented

File: nesta/production/orms/nih_es_config.json [hunk 8]
```diff
@@ -89,6 +112,9 @@
                 "id_of_continent": {
                     "type": "keyword"
                 },
+		"id_state_organisation": {
```
Comment (mindrones) [inline nesta/production/orms/nih_es_config.json]: to be indented

File: nesta/production/schemas/tier_1/aliases/health_scanner.json [hunk 9]
```diff
@@ -39,6 +43,10 @@
         "nih": "date_end_project",
         "crunchbase": "date_death_organisation"
     },
+    "funders": {
+	"nih": "terms_of_funders",
```
Comment (mindrones) [inline nesta/production/schemas/tier_1/aliases/health_scanner.json]: to be indented

File: nesta/production/orms/crunchbase_es_config.json [hunk 10]
```diff
@@ -1,129 +1,170 @@
...
+		"status_of_organisation": {
+		    "type": "keyword"
+		},
+		"textBody_summary_organisation": {
+		    "type": "text"
+		},
+		"textBody_descriptive_organisation": {
+		    "type": "text"
+		},
+		"terms_subcategory_organisation": {
```
Comment (mindrones) [inline nesta/production/orms/crunchbase_es_config.json]: this should a multiple analyzer (text, keyword)

File: nesta/production/orms/crunchbase_es_config.json [hunk 11]
```diff
@@ -1,129 +1,170 @@
...
+		    "type": "keyword"
+		},
+		"terms_category_organisation": {
```
Comment (mindrones) [inline nesta/production/orms/crunchbase_es_config.json]: this should a multiple analyzer (text, keyword)

**Comment by jaklinger** [comment]
> added comments about indentation and sorting in json files

thanks @mindrones just fixed those. I'll reprocess the `companies` data with the changes to the `mapping` for those two fields. In the meanwhile, use `companies_v1` until `v2` appears.

**Comment by jaklinger** [comment]
`companies_v2` is now available.

**Pull request merged**